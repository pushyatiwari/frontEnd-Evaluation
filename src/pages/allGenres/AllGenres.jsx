import React from 'react';
import PropTypes from 'prop-types';
import Song from '../../components/song/Song';
import './AllGenres.scss';
import Button from '../../components/button/Button';
import pop from '../../assets/genre-pop.png';
import country from '../../assets/genre-country.png';
import rock from '../../assets/genre-rock.png';
import bollywood from '../../assets/genre-bollywood.png';

const renderAllSongs = (allsongs) => {
  console.log('gensres all songs: ', allsongs);
  let i = 0;
  const renderSongs = allsongs.map((song) => {
    i += 1;
    return (
      <React.Fragment key={song.id}>
        <Song song={song} color={i % 2 === 0} />
      </React.Fragment>
    );
  });
  return renderSongs;
};
const renderAllGenres = (groupedGenres) => (
  Object.keys(groupedGenres).map((genre) => {
    let genreImage = pop;
    if (genre === 'Country') {
      genreImage = country;
    }
    if (genre === 'Rock') {
      genreImage = rock;
    }
    if (genre === 'Bollywood') {
      genreImage = bollywood;
    }
    return (
      <div className="genre-container">
        <div className="genre-title">
          <img src={genreImage} alt="genre" />
          <Button text={genre} className="genre-button" />
        </div>
        <div className="all-genre-songs">
          {renderAllSongs(groupedGenres[genre])}
        </div>
      </div>
    );
  })
);

const AllGenres = ({ groupedGenres }) => {
  console.log('inside all genres', groupedGenres);
  return (
    <>
      <div className="all-songs-container">
        <h1>genres</h1>
        {renderAllGenres(groupedGenres)}
      </div>
    </>
  );
};

AllGenres.propTypes = {
  groupedGenres: PropTypes.objectOf(PropTypes.shape).isRequired,

};
export default AllGenres;
