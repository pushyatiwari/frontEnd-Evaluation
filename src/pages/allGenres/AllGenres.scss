@import '../../style/index.scss';

.all-genre-songs{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    height: 15rem;
    width: 75rem;
    font-family: $theme-font;
}
.genre-container{
    display: flex;
    flex-wrap: wrap;
    margin-bottom: 30rem;
}

h1{
    font-family: $theme-font;
}
.genre-title{
    display: flex;
}