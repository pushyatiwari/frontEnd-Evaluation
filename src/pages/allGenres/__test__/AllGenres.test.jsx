import {
  render,
} from '@testing-library/react';
import React from 'react';
import AllGenres from '../AllGenres';

describe(AllGenres.name, () => {
  const mockGroupedGenres = {
    bollwood:
    [
      {
        id: '7ff3c998-3f7b-4b64-a764-868b58465992',
        name: 'Beautiful Mistakes (feat. Megan Thee Stallion)',
        genre: {
          id: '13082d62-0689-468a-8250-257bf2e434f3',
          name: 'Pop',
        },
        artist: { id: '2c05b7d6-f19a-4327-936e-49bfcbd9bf5c', name: 'Maroon 5' },
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273a6c1dea2b83a2309d9e0adc3',
      }],
  };
  test('renders  react link', () => {
    const app = render(<AllGenres groupedGenres={mockGroupedGenres} />);
    expect(app).toMatchSnapshot();
  });
});
