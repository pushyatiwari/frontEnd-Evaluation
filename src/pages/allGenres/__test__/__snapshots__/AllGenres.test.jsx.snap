// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGenres renders  react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="all-songs-container"
      >
        <h1>
          genres
        </h1>
        <div
          class="genre-container"
        >
          <div
            class="genre-title"
          >
            <img
              alt="genre"
              src="genre-pop.png"
            />
            <button
              class="button"
              type="button"
            >
              <b>
                bollwood
              </b>
            </button>
          </div>
          <div
            class="all-genre-songs"
          >
            <div
              class="black song-container"
            >
              <div
                class="album-art-container"
              >
                <img
                  alt="fav"
                  class="album-art"
                  src="https://i.scdn.co/image/ab67616d0000b273a6c1dea2b83a2309d9e0adc3"
                />
              </div>
              <div
                class="song-card-bottom"
              >
                <div
                  class="song-card-bottom-left"
                >
                  <p>
                    Beautiful Mistakes (feat. Megan Thee Stallion)
                  </p>
                  <p>
                    Maroon 5
                  </p>
                </div>
                <img
                  alt="fav"
                  class="fav"
                  src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRYlLHjRrXnSZtMYz4tylnrrAfSTY5A5EmyRQ&usqp=CAU"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="all-songs-container"
    >
      <h1>
        genres
      </h1>
      <div
        class="genre-container"
      >
        <div
          class="genre-title"
        >
          <img
            alt="genre"
            src="genre-pop.png"
          />
          <button
            class="button"
            type="button"
          >
            <b>
              bollwood
            </b>
          </button>
        </div>
        <div
          class="all-genre-songs"
        >
          <div
            class="black song-container"
          >
            <div
              class="album-art-container"
            >
              <img
                alt="fav"
                class="album-art"
                src="https://i.scdn.co/image/ab67616d0000b273a6c1dea2b83a2309d9e0adc3"
              />
            </div>
            <div
              class="song-card-bottom"
            >
              <div
                class="song-card-bottom-left"
              >
                <p>
                  Beautiful Mistakes (feat. Megan Thee Stallion)
                </p>
                <p>
                  Maroon 5
                </p>
              </div>
              <img
                alt="fav"
                class="fav"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRYlLHjRrXnSZtMYz4tylnrrAfSTY5A5EmyRQ&usqp=CAU"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
